### PR TITLE
Use the current credentials when trying to get the bucket region

### LINKED ISSUE #15420: China S3 bucket resource returning 401 for non-ICP accounts
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform 0.12.26
AWS Provider v3.5.0

### Affected Resource(s)

* aws_s3_bucket

### Terraform Configuration Files

```hcl
terraform {
  required_version = "0.12.26"
}

provider "aws" {
  version = "3.5.0"
  profile = "canva-network-cn"
  region = "cn-north-1"
  skip_metadata_api_check = true
}

resource "aws_s3_bucket" "b" {
  bucket = "my-tf-test-bucket-fdsajfkdslfjlk"
  acl    = "private"
}

```

### Debug Output
https://gist.github.com/michaelfoley1/06d649284ac90fe467cce2defe839ed9

### Expected Behavior
Terraform should create the bucket and exit gracefully.

### Actual Behavior
Terraform creates the bucket and errors out with the error message:
`Error: error getting S3 Bucket location: Unauthorized: Unauthorized`

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
This only affects AWS account in the Chinese partiion that do not have ICP licenses associated to them.

If a Chinese account does have an ICP licence unauthenticated HEAD requests receive a 403 response which is gracefully handled.
It an Chinese account does not have an ICP license unauthenticated HEAD requests receive a 401 response which is not gracefully handled and errors out.

### PR DESCRIPTION
**Pull request by ebabani**
This fixes https://github.com/terraform-providers/terraform-provider-aws/issues/15420 where in aws-cn using anonymous credentials will cause the Head request to return Unauthorized. That error in turn fill cause terraform bucket operations to fail.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #15420

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_s3_bucket: Use current provider credentials when fetching the bucket region
```

Output from acceptance testing in AWS Commercial:

```
$ make testacc TESTARGS='-run= TestAccAWSS3Bucket_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSS3Bucket_basic -timeout 120m
=== RUN   TestAccAWSS3Bucket_basic
=== PAUSE TestAccAWSS3Bucket_basic
=== CONT  TestAccAWSS3Bucket_basic
--- PASS: TestAccAWSS3Bucket_basic (63.20s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	65.843s
...
```

Acceptance tests fail in aws-cn due to a separate issue (`resource_aws_s3_bucket_test.go:170: Check 2/8 error: aws_s3_bucket.bucket: Attribute 'hosted_zone_id' not found`) but have verified the fix manually.

### CODE REVIEW

**Comment by ebabani** [comment]
Hi,

Anything else that needs to be done before this is reviewed? At the moment this bug blocks s3 bucket creates/updates for accounts in `aws-cn` which don't have ICP configured.

In our use case that is every account created through AWS Organizations.

**Comment by damoco** [comment]
When will this request can be merged in?

**Comment by all4innov** [comment]
Any timeline when this PR will be merged ?

**Comment by ebabani** [comment]
Hi. Any updates on this PR? As it stands terraform s3 operations in a large subset of `aws-cn` accounts will always fail.

**Review by yogeek — APPROVED** [review]

**Comment by yogeek** [comment]
Our team currently have to use the code of this PR to deploy poduction infrastructure in China so we can confirm this code is solving the issue. Thanks again @ebabani 👍 

But until this is merged, we have to recompile the provider and override the default one by the patched one...not really ideal for an automated process !
Can someone take a look and merge please ? Thank you :-)

**Comment by Qvazi141** [comment]
Hi, guys. We also cannot continue to work in China until these changes are merged. Please merged it.

**Comment by dawidmalina** [comment]
We are also blocked by this issue :(

**Review by bflad — APPROVED** [review]
Looks good, thank you for this update, @ebabani 🚀 

Output from acceptance testing:

```
--- PASS: TestAccAWSS3Bucket_acceleration (96.61s)
--- PASS: TestAccAWSS3Bucket_AclToGrant (91.15s)
--- PASS: TestAccAWSS3Bucket_basic (58.12s)
--- PASS: TestAccAWSS3Bucket_Bucket_EmptyString (57.42s)
--- PASS: TestAccAWSS3Bucket_Cors_Delete (48.59s)
--- PASS: TestAccAWSS3Bucket_Cors_EmptyOrigin (58.18s)
--- PASS: TestAccAWSS3Bucket_Cors_Update (99.78s)
--- PASS: TestAccAWSS3Bucket_disableDefaultEncryption_whenDefaultEncryptionIsEnabled (97.18s)
--- PASS: TestAccAWSS3Bucket_enableDefaultEncryption_whenAES256IsUsed (58.48s)
--- PASS: TestAccAWSS3Bucket_enableDefaultEncryption_whenTypical (59.64s)
--- PASS: TestAccAWSS3Bucket_forceDestroy (42.93s)
--- PASS: TestAccAWSS3Bucket_forceDestroyWithEmptyPrefixes (40.26s)
--- PASS: TestAccAWSS3Bucket_forceDestroyWithObjectLockEnabled (37.51s)
--- PASS: TestAccAWSS3Bucket_generatedName (56.37s)
--- PASS: TestAccAWSS3Bucket_GrantToAcl (96.56s)
--- PASS: TestAccAWSS3Bucket_ignoreTags (92.12s)
--- PASS: TestAccAWSS3Bucket_LifecycleBasic (117.55s)
--- PASS: TestAccAWSS3Bucket_LifecycleExpireMarkerOnly (95.88s)
--- PASS: TestAccAWSS3Bucket_LifecycleRule_AbortIncompleteMultipartUploadDays_NoExpiration (59.40s)
--- PASS: TestAccAWSS3Bucket_LifecycleRule_Expiration_EmptyConfigurationBlock (50.49s)
--- PASS: TestAccAWSS3Bucket_Logging (57.51s)
--- PASS: TestAccAWSS3Bucket_namePrefix (54.99s)
--- PASS: TestAccAWSS3Bucket_objectLock (63.40s)
--- PASS: TestAccAWSS3Bucket_Policy (142.88s)
--- PASS: TestAccAWSS3Bucket_Replication (164.69s)
--- PASS: TestAccAWSS3Bucket_ReplicationConfiguration_Rule_Destination_AccessControlTranslation (117.29s)
--- PASS: TestAccAWSS3Bucket_ReplicationConfiguration_Rule_Destination_AddAccessControlTranslation (110.12s)
--- PASS: TestAccAWSS3Bucket_ReplicationExpectVersioningValidationError (45.30s)
--- PASS: TestAccAWSS3Bucket_ReplicationSchemaV2 (142.38s)
--- PASS: TestAccAWSS3Bucket_ReplicationWithoutPrefix (72.68s)
--- PASS: TestAccAWSS3Bucket_ReplicationWithoutStorageClass (79.07s)
--- PASS: TestAccAWSS3Bucket_RequestPayer (96.23s)
--- PASS: TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 (53.21s)
--- PASS: TestAccAWSS3Bucket_shouldFailNotFound (38.56s)
--- PASS: TestAccAWSS3Bucket_tagsWithNoSystemTags (181.24s)
--- PASS: TestAccAWSS3Bucket_tagsWithSystemTags (228.92s)
--- PASS: TestAccAWSS3Bucket_UpdateAcl (100.08s)
--- PASS: TestAccAWSS3Bucket_UpdateGrant (147.12s)
--- PASS: TestAccAWSS3Bucket_Versioning (132.83s)
--- PASS: TestAccAWSS3Bucket_Website_Simple (138.92s)
--- PASS: TestAccAWSS3Bucket_WebsiteRedirect (142.12s)
--- PASS: TestAccAWSS3Bucket_WebsiteRoutingRules (99.74s)
--- PASS: TestAccAWSS3MultiBucket_withTags (58.18s)

--- PASS: TestAccDataSourceS3Bucket_basic (52.91s)
--- PASS: TestAccDataSourceS3Bucket_website (52.97s)
```

**Pull request merged**
